src/components/PatientsTable/PatientsTable.tsx

```diff
@@ -0,0 +1,59 @@
+//
+// This source file is part of the Stanford Biodesign Digital Health Spezi Web Health Components open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import { type Patient } from '@medplum/fhirtypes'
+import {
+  DataTable,
+  type DataTableProps,
+} from '@stanfordspezi/spezi-web-design-system/components/DataTable'
+import { createColumnHelper } from '@tanstack/table-core'
+import { PatientMenu } from './PatientMenu'
+
+interface PatientsDataTableProps
+  extends Omit<DataTableProps<Patient>, 'columns'> {
+  onRowClick?: (data: Patient) => void
+  onDelete: (data: Patient) => void
+  editRoute: string
+}
+
+export const PatientsTable = ({
+  data,
+  onRowClick,
+  onDelete,
+  editRoute,
+  ...props
+}: PatientsDataTableProps) => {
+  const columnHelper = createColumnHelper<Patient>()
+
+  const columns = [
+    { header: 'First Name', accessorKey: 'name.0.given' },
+    { header: 'Last Name', accessorKey: 'name.0.family' },
```

Comment:
Let's use `columnHelper` for all the columns if possible. This way, we get full type-safety.

Another point - `accessorKey` as string are not worth it. With TypeScript, it's better to just write an accessor function, because we get all the type hints. I am aware that `accessKey` most likely also use type checking to verify this path exists, but from my experience those create problems sooner or later.